### ISSUE TITLE
Automatic fov settings

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#470]
Author: @rmarquis (Spyhawk)
Date: 2014-01-04
Redmine Issue: 470, https://dev.etlegacy.com/issues/470

---

Default FOV value is currently 90, which is "accurate" only for 4:3 screen. A "natural" field of view for 16:9 screen is 106.27.

Idea of new feature: when set to 0, cg_fov automatically compute the "best" FOV for your resolution. Players could still set their preferred fov by manually setting cg_fov.

Horizontal FOV formula can be found here: http://en.wikipedia.org/wiki/Field_of_view_in_video_games. 

---

Related commit(s):
91e4bb66e395ce8cd668c0e9bc1517928f2f33e7 - Revert "cgame: set default fov value to 106.27 for..
 00949de02bb5c86321053eb3e8ecb8a79858ae21 - cgame: automatic fov adjustment for wide screens, ..
 27cf19d1883cf6a9d1a6e1b9462b9fba171a6530 - ui: adjusted FOV option, refs 470
 575a33143eb0e2271b2460c19f1a3ed7d84b79be - cgame: removed obsolete cg_gun_fovscale cvar, refs..

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-15T18:06:30Z

---

From Ensi: https://github.com/JACoders/OpenJK/commit/ad3cbbd4ce17be90e17580975752128c5a5d9853

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-24T22:30:26Z

---

Applied in changeset commit:00949de02bb5c86321053eb3e8ecb8a79858ae21.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2015-03-24T22:34:45Z

---

I implemented this feature without an extra cvar, as it doesn't make sense to use a lower fov than 90 anyway (or lower than 106.27 on 16:9 screens).

I also changed cg_gun_fovscale default value to 0. The reason we implemented it back in the days was that it was impossible to have a "correct" fov on 16:9 screen without having a distorted weapon (fov 90 on 4:3 == fov 106.27 on 16:9).

With this change, fov 90 is the "correct" fov for both square and wide screens. The cg_gun_fovscale cvar doesn't have a real purpose anymore and we even might consider removing it (although some people might want to use it).
